bindings/cpp/include/metadata.hpp

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#pragma once
+
+#include <cstdint>
+#include <memory>
+#include <optional>
+#include <string_view>
+
+#include "boost/date_time/posix_time/posix_time.hpp"
+#include "lib.rs.h"
+
+namespace opendal {
+namespace ffi {
+class Metadata;
+}
+
+/**
+ * @enum class EntryMode
+ * @brief The mode of the entry
+ */
+enum class EntryMode {
+  kFile = 0,
```

Comment:
Why we need to add a `k` prefix here?

Reply:
tag it as a enum，distinguish it from macro

Reply:
that totally makes sense for `enum`, but this in an `enum class` so it is namespaced which should be totally sufficient.